### ISSUE TITLE
openstack_fw_firewall_v1 ignores value_specs on POST

### ISSUE THREAD

**Issue by geofffranks** [terraform-providers/terraform-provider-openstack#66]
### Terraform Version

Terraform v0.10.0

### Affected Resource(s)

- openstack_fw_firewall_v1

### Terraform Configuration Files
```hcl
variable az2 = "us-1b"

resource "openstack_fw_policy_v1" "firewall-policy-1" {
  name = "cross-env-network-policy-1"
  rules = []
  value_specs = {
    availability_zone = "${var.az2}"
  }
}

resource "openstack_fw_firewall_v1" "firewall-1" {
  name      = "firewall-1"
  policy_id = "${openstack_fw_policy_v1.firewall-policy-1.id}"

  associated_routers = [
    "${openstack_networking_router_v2.router-1.id}",
  ]

  value_specs = {
    availability_zone = "${var.az2}"
  }
}
```

### Debug Output

https://gist.github.com/geofffranks/c1f64cf1463208607a28fcc150ade7d6

### Panic Output

None

### Expected Behavior

Terraform should have appended `"availability_zone": "us1-b"` to the HTTP POST body

### Actual Behavior

The `availability_zone` attribute was omitted from the POST body, after being listed as an attribute that would be set via the debugging.

### Steps to Reproduce
1. `terraform apply`

### Important Factoids

`openstack_fw_policy_v1` and `openstack_fw_rule_v1` objects were created properly with the `value_specs` data. Just `openstack_fw_firewall_v1` had issues.

**Comment by geofffranks** [comment]
Seems like most (all?) of the resources in https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/types.go use the BuildRequest() function. However the `ToFirewallCreateMap()` resource doesn't. Could this be the cause?

**Comment by fatmcgav** [comment]
@geofffranks You're right in that `ToFirewallCreateMap()` doesn't use `BuildRequest()`, but that's because of the router assosiation support... 

Are you able to try without `associated_routers` and see if it works?

On a tangent, I notice you're using Fujitsu K5. How are you finding it?

**Comment by fatmcgav** [comment]
@geofffranks I've managed to reproduce this failure. 

Just need to work out a fix now :)

**Comment by geofffranks** [comment]
Thanks! If there's anything I can do to help, let me know.

K5 is pretty good. Ran into a few quirks with creating routers with gateways + routes (need to take a two-phased approach). It would be nice if terraform could support their networking addons like network connectors + endpoints for cross-AZ communications, but totally understandable why it doesn't.

**Comment by fatmcgav** [comment]
@geofffranks Yeh, that router>gateway quirk caught me out for a while aswell, to the point whereby I build a custom version of TF that changed the order... Which reminds me, I was going to try and come up with a more permanent solution for that... 

WRT custom K5 resources, I might be able to help with that... 
Drop me an email to 'gavin.williams[at]weareact.com'

**Comment by geofffranks** [comment]
cc/ @xiujiao

**Comment by fatmcgav** [comment]
@geofffranks Just merged a fix for this into master - See #70.

Feel free to build from source and test, or hopefully we should be cutting a new release in the next couple of days...
